mava/adders/reverb/sequence.py

```diff
@@ -65,183 +67,83 @@ def __init__(
-        if first_write:
-            delta = self._max_sequence_length - self._step
-        else:
-            delta = (
-                self._period - (self._step - self._max_sequence_length)
-            ) % self._period
-
-        # Bump up to the position where we will write a sequence.
-        self._step += delta
-
-        if self._pad_end_of_episode:
-            zero_step = tree.map_structure(utils.zeros_like, final_step)
-
-            # Pad with zeros to get a full sequence.
-            for _ in range(delta):
-                self._buffer.append(zero_step)
-                self._writer.append(zero_step)
-        elif not first_write:
-            # Pop items from the buffer to get a truncated sequence.
-            # Note: this is consistent with the padding loop above, since adding zero
-            # steps pops the left-most elements. Here we just pop without padding.
-            for _ in range(delta):
-                self._buffer.popleft()
-
-        # Write priorities for the sequence.
-        self._maybe_add_priorities()
-
-    def _maybe_add_priorities(self) -> None:
-        if not (
-            # Write the first time we hit the max sequence length...
-            self._step == self._max_sequence_length
-            or
-            # ... or every `period`th time after hitting max length.
-            (
-                self._step > self._max_sequence_length
-                and (self._step - self._max_sequence_length) % self._period == 0
-            )
-        ):
+        if not first_write and not period_reached and not force:
             return
 
+        # TODO(b/183945808): will need to change to adhere to the new protocol.
+        if not end_of_episode:
+            get_traj = operator.itemgetter(slice(-sequence_length - 1, -1))
+        else:
+            get_traj = operator.itemgetter(slice(-sequence_length, None))
+
+        history = self._writer.history
+        trajectory = base.Trajectory(**tree.map_structure(get_traj, history))
+
         # Compute priorities for the buffer.
-        steps = list(self._buffer)
-        num_steps = len(steps)
-        table_priorities = utils.calculate_priorities(self._priority_fns, steps)
+        table_priorities = utils.calculate_priorities(self._priority_fns, trajectory)
 
         # Create a prioritized item for each table.
         for table_name, priority in table_priorities.items():
-            self._writer.create_item(table_name, num_steps, priority)
+            self._writer.create_item(table_name, priority, trajectory)
+            self._writer.flush(self._max_in_flight_items)
 
     @classmethod
     def signature(
         cls,
         environment_spec: specs.EnvironmentSpec,
-        extras_spec: tf.TypeSpec = {},
+        sequence_length: Optional[int] = None,
+        extras_spec: NestedSpec = (),
     ) -> tf.TypeSpec:
-        """This is a helper method for generating signatures for Reverb tables.
```

Comment:
Need to update the docstring here and a few other places. But I guess the docstring checker will catch this in future PRs? ;)